InfiniteRecharge/src/main/java/frc/robot/CANChecker.java

```diff
@@ -0,0 +1,57 @@
+package frc.robot;
+
+import java.util.ArrayList;
+
+import com.ctre.phoenix.motorcontrol.can.BaseTalon;
+
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+
+// to see what's connected to the CAN Bus and what's not
+public class CANChecker {
+    private final ArrayList<BaseTalon> talons;
+    private final String NAME = "CAN Bus";
+
+    private int i = -1; // 1 will get added on first iteration
+    private final int UPDATE_PERIOD = 25; // every 25 iterations, called at 50Hz = twice a second
+
+
+
+    public CANChecker() {
+        talons = new ArrayList<BaseTalon>();
+    }
+
+    public void addTalons(BaseTalon... newTalons) {
+        for (int i = 0; i < newTalons.length; i++) {
+            talons.add(newTalons[i]);
+        }
+    }
+
+    public void periodic() {
+        i++;
+        i = i % UPDATE_PERIOD;
+        if (i == 0) {
+            check();
+        }
+    }
+
+    public void check() {
+        boolean allGood = true;
+
+        for (int i = 0; i < talons.size(); i++) {
+            BaseTalon talon = talons.get(i);
+            int id = talon.getDeviceID();
+
+            int param = talon.configGetCustomParam(0);
```

Comment:
Does this work in disabled mode? It’s not terrible if it doesn’t, but we want to make sure the pit team knows they have to enable the robot for the CAN checker to work.